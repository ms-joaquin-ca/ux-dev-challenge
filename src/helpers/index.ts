export * from './styles'
export * from './tests'
