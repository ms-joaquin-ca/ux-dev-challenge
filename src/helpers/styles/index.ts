export { default as Normalize } from './normalize'
