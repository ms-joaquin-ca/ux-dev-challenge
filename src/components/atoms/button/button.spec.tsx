import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import type { ButtonProps } from 'typings/components'
import { provideTheme } from 'helpers'

import Button from './button'

const defaultProps: ButtonProps = {
  color: 'primary',
  variant: 'contained',
}

const handleClick = jest.fn()

const setup = (props: ButtonProps) =>
  render(
    provideTheme(
      <Button onClick={handleClick} {...props}>
        Test Button
      </Button>
    )
  )

describe('Button', () => {
  it('renders a default button and click on it', () => {
    const wrapper = setup(defaultProps)

    const button = wrapper.getByRole('button')

    userEvent.click(button)

    expect(handleClick).toHaveBeenCalledTimes(1)

    expect(wrapper.container).toMatchSnapshot()
  })

  it('renders an outlined secondary full width button', () => {
    const wrapper = setup({
      ...defaultProps,
      variant: 'outlined',
      color: 'secondary',
      fullWidth: true,
    })

    expect(wrapper.container).toMatchSnapshot()
  })

  it('renders a disabled button', () => {
    const wrapper = setup({ ...defaultProps, disabled: true })

    const button = wrapper.getByRole('button')

    expect(button).toBeDisabled()

    expect(wrapper.container).toMatchSnapshot()
  })
})
