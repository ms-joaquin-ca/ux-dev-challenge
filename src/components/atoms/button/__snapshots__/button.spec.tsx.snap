// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a default button and click on it 1`] = `
<div>
  <button
    class="sc-bcXHqe iBkrgz"
    color="primary"
  >
    Test Button
  </button>
</div>
`;

exports[`Button renders a disabled button 1`] = `
<div>
  <button
    class="sc-bcXHqe gLDDHg"
    color="primary"
    disabled=""
  >
    Test Button
  </button>
</div>
`;

exports[`Button renders an outlined secondary full width button 1`] = `
<div>
  <button
    class="sc-bcXHqe fJAKEf"
    color="secondary"
  >
    Test Button
  </button>
</div>
`;
